### PR TITLE
fix: security issue in account creation (regarding emails)

### PR DESCRIPTION
**Pull request by pwbs**
modified:   src/os_db.ml

### CODE REVIEW

**Comment by dannywillems** [comment]
@pwbs I think it's interesting to export the function `pg_escape_quotes` for other PGOCaml requests.

**Comment by pwbs** [comment]
`pg_escape_quotes` has disappeared... ^^' 

Do you want it to be somewhere?

**Comment by dannywillems** [comment]
**!! Don't merge: it needs a database connection to compile. !!**

**Comment by pwbs** [comment]
Indeed, I have to rewrite that using macaque :( :( :( :( :( :(

File: src/os_handlers.eliom [hunk 1]
```diff
@@ -295,7 +295,7 @@ let action_link_handler_common akey =
       else Lwt.return_unit
     in
     let%lwt () =
-      if action = `AccountActivation
+      if action = `AccountActivation || action = `PasswordReset
```
Comment (pwbs) [inline src/os_handlers.eliom]: (preventing registering an email that's already been through the registration process) => (validate emails for which we require a password reset)

**Comment by pwbs** [comment]
I've push-f-ed and it's now using macaque...

It's incredibly difficult to use macaque...

**Pull request merged**